paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -176,15 +175,34 @@ internal class IntentConfirmationHandler(
     fun start(
         arguments: Args,
     ) {
-        if (_state.value is State.Confirming) {
+        val currentState = _state.value
+
+        if (currentState is State.Preconfirming || currentState is State.Confirming) {
             return
         }
 
-        _state.value = State.Confirming(isPaymentSheetVisible = true)
+        val confirmationOption = arguments.confirmationOption
+
+        val requiresPreconfirm = confirmationOption is PaymentConfirmationOption.GooglePay ||
+            confirmationOption.isBacs()
+
+        _state.value = if (requiresPreconfirm) {
```

Comment:
I think that instead of figuring out here whether we need to call preconfirm, we should just have preconfirm do nothing for the payment methods that don't need any preconfirm actions to be done

That would simplify + centralize this logic, because rn we need to check in multiple places whether the payment option requires the preconfirm step

Reply:
I did this initially but wanted to avoid snap state updates (initially moving to `Preconfirming` then immediately to `Confirming`). It may not be a problem though given `StateFlow` does not re-emit equal state values in the UI.